.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -693,6 +693,15 @@ private DateTimeResolutionResult ParsePureNumberCases(string text, DateObject re
 
             if (match.Success && (match.Index == 0 || match.Index + match.Length == trimmedText.Length))
             {
+
+                // Just because we think we found a time period doesn't mean it is one, it could be the start of a hyphenated date
+                var hyphenDateMatch = this.Config.HyphenDateRegex.Match(trimmedText);
+
+                if (hyphenDateMatch.Success && hyphenDateMatch.Index >= match.Index && (match.Index + match.Length) <= (hyphenDateMatch.Index + hyphenDateMatch.Length))
```

Comment:
Thanks for proposing a fix, but we need to test the performance impact of this change.
Could you run some experiments?

Reply:
Could you also add some details in the PR description about the root cause you've identified.
It helps make sure the intent of the fix and as future documentation.

Reply:
> Thanks for proposing a fix, but we need to test the performance impact of this change.
> Could you run some experiments?

This PR now causes about 0.8% performance drop according to my test. :(

Reply:
I tested the DataDrivenTests in the project and

With Fix: Run finished: 20231 tests run (0:02:34.0273429) ==========
Without Fix: Run finished: 20229 tests run (0:02:52.0896418) ==========

Seem they ran a little bit faster with the fix.

Maybe I am not validating correctly?

Reply:
The test runs vary from time to time but so far they seem to run faster with the fix.
Run 2
With Fix: Run finished: 20231 tests run (0:02:25.6527977) =========
Without Fix: Run finished: 20229 tests run (0:02:31.4956609) ==========

Reply:
I re-checked it in the current master, it seems no performance drop now.